### PR TITLE
Add pull request template.

### PR DESCRIPTION
**Pull request by r0mant**
As discussed today, add pull request template. Should we add anything else to it?

Testing on my own repo it also looks like this should only go to the master branch, so version branches don't require it.

### CODE REVIEW

**Comment by knisbet** [comment]
I would consider structuring the template more as a checklist or including checklists as part of the request to help communicate the status of a particular PR. May also help with creating a standard that easier to evolve over time.


Example:
- https://github.com/coreos/flannel/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Here's some references for the type of information captured in other OSS projects:
- https://github.com/kubernetes/kubernetes/blob/master/.github/PULL_REQUEST_TEMPLATE.md
- https://embeddedartistry.com/blog/2017/08/04/a-github-pull-request-template-for-your-projects/

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -0,0 +1,11 @@
+## Description
+<!--Provide high-level overview of what the PR is for.-->
+
+## Implementation
+<!--Add any relevant implementation details that might help the reviewers.-->
```
Comment (wadells) [inline .github/pull_request_template.md:5]: For my edification, could you link to an PR that is a good example of providing implementation details?  I don't presently understand what exactly should go in this section (or if it should be present on most/all PRs)

**Review by wadells — COMMENTED** [review]
> Testing on my own repo it also looks like this should only go to the master branch, so version branches don't require it.

Under the theory of "Show, don't tell" will you please link to a PR from this template in your repo?

**Comment by r0mant** [comment]
@knisbet @wadells Updated to include more info and a checklist. Here's an example of a PR created using this template: https://github.com/r0mant/emacs.d/pull/5. Note that the PR was not made to a master branch (roman/test1<--roman/test2) but still uses the template like I mentioned.

File: .github/pull_request_template.md (hunk 1 above)
Comment (r0mant) [reply .github/pull_request_template.md:34]: Sure, this is basically to have a chance to explain any implementation details that you think may not be immediately obvious to the reviewer, or help understand the high-level approach to implementation of larger changesets, etc. For example, look at https://github.com/gravitational/gravity/pull/1317 (implementation section is sort of woven into the description there) or https://github.com/gravitational/gravity/pull/1084 or https://github.com/gravitational/gravity/pull/898, etc.

**Review by wadells — APPROVED** [review]
Overall, I think this feels a bit "heavy".  However, I also I think we should merge it and give it a try for a week or two.  We can always tweak it later to add missing parts or remove sections that aren't regularly providing value.

**Comment by r0mant** [comment]
Here's an example of "real" PR using this template: https://github.com/gravitational/gravity/pull/1453. I agree it feels a bit heavy. Maybe instead of checkboxes we could use bullet points in some sections. Or utilize Github labels for some parts instead?

**Comment by knisbet** [comment]
I think for "Type of change" should also be remove those that don't apply.

**Review by a-palchikov — APPROVED** [review]

**Review by UnderGreen — APPROVED** [review]

**Review by bernardjkim — APPROVED** [review]

**Pull request merged**